src/BoundaryConditions/BoundaryConditions.jl

```diff
@@ -3,8 +3,8 @@ module BoundaryConditions
 export
     BCType, Flux, Gradient, Value, Open,
     BoundaryCondition, getbc, setbc!,
-    PeriodicBoundaryCondition, OpenBoundaryCondition, NoFluxBoundaryCondition, CommunicationBoundaryCondition,
-    FluxBoundaryCondition, ValueBoundaryCondition, GradientBoundaryCondition,
+    PeriodicBoundaryCondition, OpenBoundaryCondition, NoFluxBoundaryCondition, CommunicationBoundaryCondition, 
+    FluxBoundaryCondition, ValueBoundaryCondition, GradientBoundaryCondition, HaloCommunicationBoundaryCondition,
```

Comment:
What's the difference between "CommunicationBoundaryCondition" and "HaloCommunicationBoundaryCondition"

Reply:
One is for shared communication (MultiRegion) the other for distributed communication (DistributedGrid). Probably we can rename them to reflect this difference

Reply:
So they both communicate halos?

Reply:
I think if they both communicate the same thing then one way to distinguish them would be `MultiRegionCommunicationBoundaryCondition` and `DistributedCommunicationBoundaryCondition`. That names them after the modules they refer to, I think

Reply:
Ok, I'll wait to have a commit with all tests passing, then I will do all the refactoring. So I know I have a commit to revert to if I mess up